### PR TITLE
Add Erlang Probe

### PR DESCRIPTION
**Pull request by jvanbaarsen**
Currently we support the following set of data:

* IO, tagged as Input, and Output
* Scheduler Metrics, tagged as Total, and Online
* Process metrics, tagged as Limit, and Count
* Memory, tagged as: total, processes, processes_used, system, atom,
  atom_used, binary, code, ets

In the future we might want to expand this with Scheduler utilization
and maybe more memory allocations.

You can use the following YAML for the dashboard:
```yaml
-
  title: 'Erlang Metrics'
  graphs:
    -
      title: IO
      line_label: '%kind%'
      kind: gauge
      fields:
        - erlang_io
      tags:
        kind: '*'
    -
      title: Schedulers
      line_label: '%kind%'
      kind: gauge
      fields:
        - erlang_schedulers
      tags:
        kind: '*'
    -
      title: Processes
      line_label: '%kind%'
      kind: gauge
      fields:
        - erlang_processes
      tags:
        kind: '*'
    -
      title: memory
      line_label: '%kind%'
      kind: gauge
      fields:
        - erlang_memory
      tags:
        kind: '*'

```

### CODE REVIEW

File: lib/appsignal/probes/erlang_probe.ex [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    Appsignal.set_gauge(
+      "erlang_schedulers",
+      :erlang.system_info(:schedulers_online),
+      %{kind: "Online"}
+    )
+  end
+
+  defp process_metrics() do
+    Appsignal.set_gauge("erlang_processes", :erlang.system_info(:process_limit), %{kind: "Limit"})
+    Appsignal.set_gauge("erlang_processes", :erlang.system_info(:process_count), %{kind: "Count"})
```
Comment (tombruijn) [inline lib/appsignal/probes/erlang_probe.ex]: we probably want to lowercase tag values for consistency

Comment (jvanbaarsen) [reply lib/appsignal/probes/erlang_probe.ex]: 👍

**Comment by tombruijn** [comment]
@jvanbaarsen can you make a PR on the [public_config](https://github.com/appsignal/public_config) repo for the dashboard?

File: lib/appsignal/probes/erlang_probe.ex [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  def call do
+    io_metrics()
+    scheduler_metrics()
+    process_metrics()
+    memory_metrics()
+  end
+
+  defp io_metrics do
+    {{:input, input}, {:output, output}} = :erlang.statistics(:io)
+    Appsignal.set_gauge("erlang_io", Kernel.div(input, 1024), %{kind: "input"})
```
Comment (tombruijn) [inline lib/appsignal/probes/erlang_probe.ex]: Can we rename these tag keys to `type`?
```suggestion
    Appsignal.set_gauge("erlang_io", Kernel.div(input, 1024), %{type: "input"})
```

Context, also here: https://github.com/appsignal/appsignal-ruby/pull/488#discussion_r265551406

**Comment by jvanbaarsen** [comment]
Well this broke in some weird ways after we first merged in master, and then rebased the branch this was pointed at. I'm gonna cherry pick all the changes, and create a new PR with those changes.

**Comment by jvanbaarsen** [comment]
Superseded by #466

**Pull request closed without merging**